File: Treasury.Domain/Model/FundingAccount.cs
﻿using System;
using System.Collections.Generic;

#nullable disable

namespace Treasury.Domain.Model
{
    public partial class FundingAccount
    {
        public string AccountName { get; set; }
        public string FiscalYear { get; set; }
        public decimal? FallTransfer { get; set; }
        public decimal? SpringTransfer { get; set; }
        public string WorkDayCode { get; set; }
        public DateTime Timestamp { get; set; }
    }
}

File: src/main/java/edu/wpi/sga/treasury/domain/model/FundingAccount.java
package edu.wpi.sga.treasury.domain.model;

import lombok.Getter;
import lombok.Setter;

import javax.persistence.*;
import java.math.BigDecimal;
import java.time.LocalDateTime;
import java.util.LinkedHashSet;
import java.util.Set;

@Getter
@Setter
@Entity
@Table(name = "funding_account")
public class FundingAccount {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id", nullable = false)
    private Integer id;

    @Column(name = "account_name", nullable = false)
    private String accountName;

    @Column(name = "fiscal_year", nullable = false)
    private String fiscalYear;

    @Column(name = "fall_transfer", precision = 10, scale = 2)
    private BigDecimal fallTransfer;

    @Column(name = "spring_transfer", precision = 10, scale = 2)
    private BigDecimal springTransfer;

    @Column(name = "worktag", nullable = false, length = 8)
    private String worktag;

    @Column(name = "last_modified", nullable = false)
    private LocalDateTime lastModified;

    @OneToMany(mappedBy = "fundingAccountFrom")
    private Set<Transfer> fromTransfers = new LinkedHashSet<>();

    @OneToMany(mappedBy = "fundingAccountTo")
    private Set<Transfer> toTransfers = new LinkedHashSet<>();

}